core/ts/Poll.ts

```diff
@@ -554,21 +554,62 @@ export class Poll implements IPoll {
+                // which sends a message that when force decrypted on the circuit
+                // results in a valid state index thus forcing the circuit to look
+                // for a valid state leaf, and failing to generate a proof
+
+                // gen shared key
+                const sharedKey = Keypair.genEcdhSharedKey(this.coordinatorKeypair.privKey, encPubKey);
+
+                // force decrypt it
+                const { command } = PCommand.decrypt(message, sharedKey, true);
+
+                // cache state leaf index
+                const stateLeafIndex = command.stateIndex;
+
+                // if the state leaf index is valid then use it
+                if (stateLeafIndex < this.stateLeaves.length) {
+                  currentStateLeaves.unshift(this.stateLeaves[Number(stateLeafIndex)].copy());
+                  currentStateLeavesPathElements.unshift(this.stateTree!.genProof(Number(stateLeafIndex)).pathElements);
+
+                  // copy the ballot
+                  const ballot = this.ballots[Number(stateLeafIndex)].copy();
+                  currentBallots.unshift(ballot);
+                  currentBallotsPathElements.unshift(this.ballotTree!.genProof(Number(stateLeafIndex)).pathElements);
+
+                  // add the first vote of this ballot
+                  currentVoteWeights.unshift(ballot.votes[0]);
+
+                  // create a new quinary tree and add all votes we have so far
+                  const vt = new IncrementalQuinTree(this.treeDepths.voteOptionTreeDepth, 0n, STATE_TREE_ARITY, hash5);
+
+                  // fill the vote option tree with the votes we have so far
+                  for (let j = 0; j < this.ballots[0].votes.length; j += 1) {
+                    vt.insert(ballot.votes[j]);
+                  }
+
+                  // get the path elements for the first vote leaf
+                  currentVoteWeightsPathElements.unshift(vt.genProof(0).pathElements);
+                } else {
+                  // just use state leaf index 0
+                  currentStateLeaves.unshift(this.stateLeaves[0].copy());
+                  currentStateLeavesPathElements.unshift(this.stateTree!.genProof(0).pathElements);
+                  currentBallots.unshift(this.ballots[0].copy());
+                  currentBallotsPathElements.unshift(this.ballotTree!.genProof(0).pathElements);
+
+                  // Since the command is invalid, we use a zero vote weight
+                  currentVoteWeights.unshift(this.ballots[0].votes[0]);
+
+                  // create a new quinary tree and add an empty vote
+                  const vt = new IncrementalQuinTree(this.treeDepths.voteOptionTreeDepth, 0n, STATE_TREE_ARITY, hash5);
+                  vt.insert(this.ballots[0].votes[0]);
+                  // get the path elements for this empty vote weight leaf
+                  currentVoteWeightsPathElements.unshift(vt.genProof(0).pathElements);
```

Comment:
have a question here: if a voter sent an invalid message, will the whole tree being updated (and insert the empty vote?) --> if yes, would that cause a tree full of empty votes since someone keep on sending invalid message?

Reply:
this is just a local tree generated to hold the current user (stateIndex) vote. in this case, the vote was completely invalid, so we just send dummy data which does not prevent a proof from being generated, but really has no effect in the state machine logic. this.ballots[0] will be the ballot of the stateIndex 0 which is the empty state leaf we put in the tree, so it'll just be 0